### DISCUSSION TITLE
OPTIMIZED_EXTERNAL_IMAGE_REQUEST_UNAUTHORIZED

### DISCUSSION THREAD

**Discussion by kevinmitch14** [vercel/vercel#8519] [Help]
Hey, 

I am having issues using next Image component and Google APIs. I have recently implemented the Google Autocomplete service instead of the Google Places Details. Originally with the Place Details I was getting a photo_reference which I could use to render an image with Next/Image. 

Since moving over to Autocomplete, I no longer get the `photo_reference` but I get a function called `getUrl()`, which gets an image URL corresponding to the specified options. This works as expected, but when using Next/Image anywhere other than localhost, I get the `OPTIMIZED_EXTERNAL_IMAGE_REQUEST_UNAUTHORIZED` error. 

I have allowed domains as the following `['maps.googleapis.com', "cdn.shopify.com"]` so I don't think that is the issue.

This is the two approaches, which return the exact same thing, [this](https://lh3.googleusercontent.com/places/AKR5kUhNA-HPsnhm7ZY1JBzeLVzZIsSvcXSQ99GJjneSieZPetSg6NjqFMx27aD1xSX2c0rrgf0n1WBVV5R9uL3ucmPFq8e9o9_IsY0=s1600-w3024-h4032). photo_reference version works with Next/Image but getUrl version does not. I am quite confused by this so if anybody has an idea why/how this is happening I would be so grateful.

**getUrl()**: https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAeJbb3dh6cUx9ZBADsceeR8UF42OCpf8KlVQigmV0_0mVYYjOt57MpelNrG0rTQEEs-yhpi4yIImhF50LZPTer0Ani4AZVFVY7UQFM8qJUen9QFvaQGUqi5GCqs_VnFsoeqb3UPVlTBfQvNmJBaeLe0iLlmnC0ozzmJ3v2y0sc9fSfT2eT_K&3u3024&4u4032&5m1&2e1&callback=none&key={API_KEY}&token=56237

**photo_reference** approach: https://maps.googleapis.com/maps/api/place/photo?maxwidth=3024&maxheight=4032&photo_reference=AeJbb3cAQJMQ1bRVf1OuF5VvDoEe38bKDAWAtJ4sQJZMsg1sHmhsh1BOqea5ibRomn75tAcfrHv1RgQBh18U8XrpY8H5Jm6p9NmxzQ9yGKdsFXmlkNHirG6nHyxPW0jH6eE3O5x88YIR76EZhesqxKaqR4Xqqg10uqUYM3hej74g0LVHItZS&key={API_KEY}

**Comment by amyegan** [comment]
Hi @kevinmitch14! Both of your example links return [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) response for me. The first thing that comes to mind is that there might be something you need to adjust in your Project's [environment variables](https://vercel.com/docs/concepts/projects/environment-variables). Maybe Autocomplete needs a new token with different permissions?

**Reply by kevinmitch14** [reply]
Hey @amyegan thanks for your response, I redacted my API key for both examples given, when the API key is present the examples work perfectly fine. I've just resorted to using an <img> tag instead on Next/Image but I would like to get Next/Image implemented! 

Here is an example, https://cafe-comprende-9gi82as9p-kevinmitch14.vercel.app/, if you search for a cafe and select one you will then see one image working and the other not.

**Reply by amyegan** [reply]
Thanks for sharing that example! The `OPTIMIZED_EXTERNAL_IMAGE_REQUEST_UNAUTHORIZED` error means the image provider returned a 401 or 403 status code. Something is preventing your Vercel Project from access the image. If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or your repo, I would be happy to take a closer look to help you identify the problem.

**Reply by kevinmitch14** [reply]
Thanks Amy!

Repo: https://github.com/kevinmitch14/cafe-comprende/tree/vercel_repro
Preview link: https://cafe-comprende-32tpxejes-kevinmitch14.vercel.app/

You can see the images failing in the network tab on the preview link.

**Reply by amyegan** [reply]
Sorry for my late reply! When I directly visit the URL for the failing image, I do get a [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403). Seems like something is going wrong with your getUrl() setup. Since it's a 403 Forbidden status, it still seems like a token or key problem is most likely. 

<img width="1077" alt="image" src="https://user-images.githubusercontent.com/8247105/189724266-144a1f10-f5ac-4590-9bee-a73d14303937.png">

**Reply by kevinmitch14** [reply]
Hey @amyegan, thanks for the reply. I think this is because I hardcoded a URL there with a token that must of expired.

If you use the input and search for a cafe, you will get a fresh token, and when you directly visit the URL response for the request, you will be able to see the image. You will be able to see the OPTIMIZED_EXTERNAL_IMAGE_REQUEST_UNAUTHORIZED error then also on my preview link (Error 502)

**Reply by amyegan** [reply]
I know that the `OPTIMIZED_EXTERNAL_IMAGE_REQUEST_UNAUTHORIZED` error means the image provider returned a 401 or 403 status code to your app. Without being able to debug the code and see what's actually happening, I can't say for sure what is the cause of the problem.

You will be able to get specific guidance about Google services in the [Google Developer Community](https://developers.google.com/maps/developer-community) and they might be able to identify what is causing the image to not load. I would recommend reaching out there to see if they have any ideas.

**Reply by kevinmitch14** [reply]
Hey Amy, strange because when I use a regular <img> tag the image loads fine and when I follow the URL the image loads fine. It seems to be stemming from using next/image
